clar2wasm/src/lib.rs

```diff
@@ -87,7 +86,7 @@ pub fn compile(
         }
     };
 
-    let generator = WasmGenerator::new(&mut contract_analysis);
+    let generator = WasmGenerator::new(contract_analysis.clone());
```

Comment:
This seems like the most controversial part of the change, since this is potentially a large object to clone. I look at this as a tradeoff where we would be sacrificing some runtime performance to improve the compiler-writer's experience.

In compilers, we generally should accept adding complexity to the compiler  if it results in better performance of the compiled code. Increasing compile time to get better generated code, or cleaner compiler implementation is usually acceptable, definitely in an ahead-of-time compiler, but the tradeoff here gets trickier for a just-in-time compiler. I think our model is sort of in-between those two, where contract deployment is more like a JIT situation, but contract calls after that are like an AOT situation.

Lot of stream of consciousness there, but in the end, I think this should be okay.

Reply:
I would say this belongs to things we can re-optimize later.